src/libraries/System.Private.CoreLib/src/System/Threading/Interlocked.cs

```diff
@@ -83,6 +83,22 @@ public static IntPtr Exchange(ref IntPtr location1, IntPtr value)
             return (IntPtr)Interlocked.Exchange(ref Unsafe.As<IntPtr, int>(ref location1), (int)value);
 #endif
         }
+
+        /// <summary>Sets a platform-specific handle or pointer to a specified value and returns the original value, as an atomic operation.</summary>
+        /// <param name="location1">The variable to set to the specified value.</param>
+        /// <param name="value">The value to which the <paramref name="location1"/> parameter is set.</param>
+        /// <returns>The original value of <paramref name="location1"/>.</returns>
+        /// <exception cref="NullReferenceException">The address of location1 is a null pointer.</exception>
+        [CLSCompliant(false)]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static UIntPtr Exchange(ref UIntPtr location1, UIntPtr value)
+        {
+#if TARGET_64BIT
+            return (UIntPtr)Interlocked.Exchange(ref Unsafe.As<UIntPtr, long>(ref location1), (long)value);
```

Comment:
nit: Use `nuint` so this is an identity conversion and doesn't go through the user-defined cast operator.

Reply:
I thought the plan was the C# compiler will soon start treating nint/IntPtr and nuint/UIntPtr as pure synonyms, at which point there shouldn't be such a difference.  Have I misunderstood?

I'm fine updating this if there's actually benefit, but for now I just stuck with consistency with the existing IntPtr overloads.

Reply:
That's the plan yes. 

No strong preference on it actually happening if we'd rather wait for the compiler changes, just wanted to call it out as a current inefficiency.